src/psyclone/parse/module_info.py

```diff
@@ -92,6 +103,44 @@ def __init__(self, name, filename):
         # each module, indexed by the module names: Dict[str, Set(str)].
         self._used_symbols_from_module = None
 
+        # This variable will be a set that stores the name of all routines
+        # (based on fparser), # so we can test is a routine is defined
+        # without having to convert the AST to PSyIR. It is initialised with
+        # None so we avoid trying to parse a file more than once (parsing
+        # errors would cause routine_names to be empty, so we can test
+        # if routine_name is None vs if routine_names is empty)
+        self._routine_names = None
+
+        # This dictionary stores the mapping of routine name to the list
+        # of PSyIR Routine nodes. It is a list since in case of a generic
+        # interface several Routine nodes are required.
+        self._psyir_of_routines = None
+
+        # This map contains the list of routine names that are part
+        # of the same generic interface.
+        self._generic_interfaces = {}
+
+        # This is a dictionary that will cache non-local symbols used in
```

Comment:
Again, the PSyIR symbol table already has this functionality (or easily could have), albeit without the caching.

Reply:
Removed.